src/spikeinterface/core/numpyextractors.py

```diff
@@ -146,22 +170,31 @@ def from_times_labels(times_list, labels_list, sampling_frequency, unit_ids=None
         labels_list = [np.asarray(e) for e in labels_list]
 
         nseg = len(times_list)
+
         if unit_ids is None:
             unit_ids = np.unique(np.concatenate([np.unique(labels_list[i]) for i in range(nseg)]))
 
-        sorting = NumpySorting(sampling_frequency, unit_ids)
+        spikes = []
         for i in range(nseg):
-            units_dict = {}
             times, labels = times_list[i], labels_list[i]
-            for unit_id in unit_ids:
-                mask = labels == unit_id
-                units_dict[unit_id] = times[mask]
-            sorting.add_sorting_segment(NumpySortingSegment(units_dict))
+            unit_index = np.zeros(labels.size, dtype="int64")
+            for u, unit_id in enumerate(unit_ids):
+                unit_index[labels == unit_id] = u
+            spikes_in_seg = np.zeros(len(times), dtype=minimum_spike_dtype)
+            spikes_in_seg["sample_index"] = times
+            spikes_in_seg["unit_index"] = unit_index
+            spikes_in_seg["segment_index"] = i
+            order = np.argsort(times)
+            spikes_in_seg = spikes_in_seg[order]
+            spikes.append(spikes_in_seg)
+        spikes = np.concatenate(spikes)
+
+        sorting = NumpySorting(spikes, sampling_frequency, unit_ids)
 
         return sorting
 
     @staticmethod
-    def from_dict(units_dict_list, sampling_frequency) -> "NumpySorting":
+    def from_unit_dict(units_dict_list, sampling_frequency) -> "NumpySorting":
```

Comment:
agreed, we should at least deprecate it...

Reply:
No, the to_dict() was a huge mistake because it overwrited the to_dict() to dump the object.
It was use only internally.
So I think it is OK to get it back without transitory process

Reply:
And also now the NumpySorting is dumpable so we really need the from_dict()/to_dict() to have the true behavior that come from the base.

Reply:
Yeah, I remember I was confused about this a month ago. This is the right call. It is the current method that is overloaded with the other `to_dict` as you were telling us.